### ISSUE TITLE
Issue with long filenames in legend

### ISSUE THREAD

**Issue by toqduj** [SasView/sasview#2012]
I've just come across this issue on Windows using 5.0.4. These measurements have long filenames, that generate legends just slightly larger than the plot window box, and it seems that that messes up the plot itself. Plots are a fraction of the full size of the box and tend to shrink further when interacting with that box (see screenshot). 

Disabling the label from the right-click menu resolves this, but it reappears when the next fit is run (legend state is not maintained for plots). 

This could be addressed in a plotting refactoring together with #1677 

![image](https://user-images.githubusercontent.com/5449929/151978605-f45aa5b6-7f83-4a11-9e3d-0fd56f2a08a6.png)

**Comment by RichardHeenan** [comment]
Yes, this has been noted before, but not explicitly ticketed, it was once suggested that this is a "feature" of matplotlib, that would be tricky to solve. However I note that Mantid has managed to ignore the parts of long titles outside of the axis area, on either zooming in or else dragging the plot area smaller, which is better, but still not great. e.g. plot below where I renamed the workspace to have a very long name.
![long_title](https://user-images.githubusercontent.com/4226214/152000025-4916a73f-25cf-491b-a13a-ac2d5f28aac5.png)
.

**Comment by RichardHeenan** [comment]
I'm just trying 5.0.5rc2 on Windows, where sasview  legend boxes are being resized when changing the width of the plot, when they have a name of modest length. However if the workspace has a very long name to start with, then see plot issues as reported by Brian. (I loaded a workspace, then used right click, rename. Seems you can only rename before sending to a fit, not after). So unless there is a quick way to do what Mantid does so at least some sort of plot appears, I suspect a truncation is needed, then eventually a flag in the loader to complain when titles are too long.

**Comment by toqduj** [comment]
From the look of the screenshots, it appears that the legend in Mantid is part of the plot axes, whereas for SasView it's part of the Figure. Perhaps that's where the problem arises from?

**Comment by rozyczko** [comment]
The legend is a part of the plot axes.


in Plotter.py, l. 254:
```
self.legend = ax.legend(loc='upper right', shadow=True, prop={'size':width})
```
where 
```
ax = self.figure.add_subplot(self.current_plot)
```
with `add_subplot` returning axes of that plot

**Comment by pbeaucage** [comment]
Fixed in #2264 with additional usability workaround in #2266.
